apps/omg_watcher/lib/omg_watcher/block_getter/supervisor.ex

```diff
@@ -1,45 +0,0 @@
-# Copyright 2018 OmiseGO Pte Ltd
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-defmodule OMG.Watcher.BlockGetter.Supervisor do
-  @moduledoc """
-   This supervisor takes care of BlockGetter and State processes.
-   In case one process fails, this supervisor's role is to restore consistent state
-  """
-  use Supervisor
-  use OMG.LoggerExt
-
-  def start_link do
-    Supervisor.start_link(__MODULE__, :ok, name: __MODULE__)
-  end
-
-  def init(:ok) do
-    # State and Block Getter are linked, because they must restore their state to the last stored state
-    # If Block Getter fails, it starts from the last checkpoint while State might have had executed some transactions
-    # such a situation will cause error when trying to execute already executed transaction
-    children = [
-      {OMG.State, []},
-      %{
-        id: OMG.Watcher.BlockGetter,
-        start: {OMG.Watcher.BlockGetter, :start_link, [[]]},
-        restart: :transient
-      }
-    ]
-
-    opts = [strategy: :one_for_all, max_restarts: 1000, max_seconds: 60]
```

Comment:
Awesome that you spotted this!
I'll simply link them and links are bidirectional...

Reply:
ah, linking wasn't enough, because there's also a check in getter's `handle_continue` which double-checks whether state is clean and good to go.

If you just link, it might cause the restarts and the kills resulting from the link intertwine. What we want is to
1/ whenever any of the pair dies, the other dies
2/ State starts (this might take some time as it does some slow operation to boot up)
3/ Getter starts and can do the double check asking State

wait, now I see that after the split `init`-`handle_continue` this no longer will work as I described (even with the Supervisor), since State's "start" would be super quick and not wait for the slow operation to finish. So anyway, a proper fix will require more verbose handling

Reply:
Interesting! Is the behaviour the same for ch-ch?

Reply:
nope, this comes in with the `BlockGetter`, watcher specific.

Reply:
Got a lecture in PV! Awesome, tnx!